packages/scandipwa/src/component/ProductCard/ProductCard.container.js

```diff
@@ -87,16 +93,32 @@ export class ProductCardContainer extends PureComponent {
     });
 
     _getLinkTo() {
-        const { product: { url }, product } = this.props;
+        const {
+            base_link_url,
+            product_use_categories,
+            product: { url, url_rewrites },
+            product
+        } = this.props;
+        const { pathname: storePrefix } = new URL(base_link_url || window.location.origin);
+        const { location: { pathname } } = history;
 
         if (!url) {
             return undefined;
         }
 
         const { parameters } = this._getConfigurableParameters();
 
+        const productUrl = `${pathname.replace(storePrefix, '')}/${url.replace(storePrefix, '')}`;
+
+        // if 'Product Use Categories' is enabled then use the current window location to see if the product
+        // has any url_rewrite for that path. (if not then just use the default url)
+        const rewriteUrl = url_rewrites.filter((e) => (e.url.includes(productUrl) ? e.url : null));
```

Comment:
It seems that it's not a case for `filter`, rather `find`.
Also, you could've just `.find(({url}) => url.includes(productUrl))`!
Let's be more concise here 👍🏻